dimod/decorators.py

```diff
@@ -176,3 +179,63 @@ def new_f(*args, **kwargs):
         return new_f
 
     return _vartype_arg
+
+
+def graph_argument(*arg_names):
```

Comment:
Too bad `vartype_argument` code had to be duplicated here.

Reply:
we would look at making a generic arg coercer decorator, maybe if we get a third